src/spotted/handlers/report_spot.py

```diff
@@ -52,6 +52,10 @@ async def report_spot_callback(update: Update, context: CallbackContext) -> int:
     abusive_message_id = info.message.reply_to_message.message_id if Config.post_get("comments") else info.message_id
 
     report = Report.get_post_report(user_id=info.user_id, channel_id=info.chat_id, c_message_id=abusive_message_id)
+    user = User(info.user_id)
+    if user.is_banned:
+        await info.answer_callback_query(text="Sei stato bannato, non puoi segnalare post")
+        return ConversationState.END.value
     if report is not None:  # this user has already reported this post
         await info.answer_callback_query(text="Hai già segnalato questo spot")
```

Comment:
while you are here, to be more coherent (for consistency) use `post` instead of `spot`, or do the opposite in your changes